### ISSUE TITLE
ValueError: query data dimension must match training data dimension

### ISSUE THREAD

**Issue by Aaryanverma** [interpretml/DiCE#270]
while generating counterfactuals with 'genetic' method, the following error appears:

`ValueError: query data dimension must match training data dimension`

```
Traceback:
File "C:\Users\Anaconda3\envs\xyz\lib\site-packages\streamlit\script_runner.py", line 379, in _run_script
    exec(code, module.__dict__)
File "C:\Users\Downloads\myproject\counterfact_explain.py", line 102, in <module>
    desired_class=desired_var.item())
File "C:\Users\Anaconda3\envs\xyz\lib\site-packages\dice_ml\explainer_interfaces\explainer_base.py", line 102, in generate_counterfactuals
    **kwargs)
File "C:\Users\Anaconda3\envs\xyz\lib\site-packages\dice_ml\explainer_interfaces\dice_genetic.py", line 289, in _generate_counterfactuals
    sparsity_weight, diversity_weight, categorical_penalty, verbose)
File "C:\Users\Anaconda3\envs\xyz\lib\site-packages\dice_ml\explainer_interfaces\dice_genetic.py", line 208, in do_param_initializations
    query_instance, query_instance_df_dummies, verbose)
File "C:\Users\Anaconda3\envs\xyz\lib\site-packages\dice_ml\explainer_interfaces\dice_genetic.py", line 190, in do_cf_initializations
    indices = self.KD_tree.query(query_instance_df_dummies, num_queries)[1][0]
File "sklearn\neighbors\_binary_tree.pxi", line 1338, in sklearn.neighbors._kd_tree.BinaryTree.query
```

Dataset used : adult income dataset
Checked dimensions of both training data and query data, both are the same.
However, with 'random' method, it works perfectly.

Pls, help with this issue.

**Comment by gaugup** [comment]
@amit-sharma, I see the same issue with one of the github issues in responsible-ai-toolbox at https://github.com/microsoft/responsible-ai-toolbox/issues/1151. Attached is the notebook code which reproduces this issue. 

```
import shap
from sklearn.model_selection import train_test_split

X, y = shap.datasets.adult()
print("Data fetched")
target_feature = "income"
y = [1 if y_i else 0 for y_i in y]

full_data = X.copy()
full_data[target_feature] = y

data_train, data_test = train_test_split(
    full_data, test_size=1000, random_state=96132, stratify=full_data[target_feature]
)
```

**Comment by amit-sharma** [comment]
Let me take a look.  @soundarya98 do you have an answer to this question?

**Comment by amit-sharma** [comment]
Thanks @Aaryanverma for flagging this. This error was caused when the `query_instance` had a categorical column which was numerically encoded and so it was assumed numeric. This will be fixed now with PR #274
